packages/tools/src/tools/artifact-generator.ts

```diff
@@ -0,0 +1,148 @@
+import { Settings, type JSONValue } from "@llamaindex/core/global";
+import type { ChatMessage } from "@llamaindex/core/llms";
+import { tool } from "@llamaindex/core/tools";
+import { z } from "zod";
+
+const ARTIFACT_GENERATION_PROMPT = `You are a highly skilled content creator and software engineer. Your task is to generate either a code artifact or a document artifact based on the user's request.
```

Comment:
how about separating this into two artifact tools: one for code and one for documents